lib/miq_automation_engine/service_models/miq_ae_service_host.rb

```diff
@@ -65,14 +59,7 @@ def ems_custom_get(key)
     end
 
     def ems_custom_set(attribute, value)
-      MiqQueue.put(
-        :class_name  => @object.class.name,
-        :instance_id => @object.id,
-        :method_name => 'set_custom_field',
-        :zone        => @object.my_zone,
-        :role        => 'ems_operations',
-        :args        => [attribute, value]
-      ) if @object.is_vmware?
+      sync_or_async_ems_operation(false, 'set_custom_field', [attribute, value]) if @object.is_vmware?
```

Comment:
I would suggest moving this into a VMware subclassed service model and drop the `if @object.is_vmware?` check.